internal/config/config.go

```diff
@@ -2,152 +2,223 @@ package config
-	ErrConfigNotFound = fmt.Errorf("config not found")
-	// ErrConfigNotParsed is returned on load if the config could not be decoded.
-	ErrConfigNotParsed = fmt.Errorf("config not parseable")
+	envPrefix    = "GOPASS_CONFIG_"
+	systemConfig = "/etc/gopass/config"
 )
 
-// Config is the current config struct.
-type Config struct {
-	AutoClip      bool              `yaml:"autoclip"`      // decide whether passwords are automatically copied or not.
-	AutoImport    bool              `yaml:"autoimport"`    // import missing public keys w/o asking.
-	ClipTimeout   int               `yaml:"cliptimeout"`   // clear clipboard after seconds.
-	ExportKeys    bool              `yaml:"exportkeys"`    // automatically export public keys of all recipients.
-	NoPager       bool              `yaml:"nopager"`       // do not invoke a pager to display long lists.
-	Notifications bool              `yaml:"notifications"` // enable desktop notifications.
-	Parsing       bool              `yaml:"parsing"`       // allows to switch off all output parsing.
-	Path          string            `yaml:"path"`
-	SafeContent   bool              `yaml:"safecontent"` // avoid showing passwords in terminal.
-	Mounts        map[string]string `yaml:"mounts"`
-	UseKeychain   bool              `yaml:"keychain"` // use OS keychain for age
+func newGitconfig() *gitconfig.Configs {
+	c := gitconfig.New()
+	c.EnvPrefix = envPrefix
+	c.GlobalConfig = os.Getenv("GOPASS_CONFIG")
+	c.SystemConfig = systemConfig
+
+	return c
+}
 
-	ConfigPath string `yaml:"-"`
+var defaults = map[string]string{
+	"core.autosync":      "true",
+	"core.cliptimeout":   "45",
+	"core.exportkeys":    "true",
+	"core.notifications": "true",
+	"core.parsing":       "true",
+}
 
-	// Catches all undefined files and must be empty after parsing.
-	XXX map[string]any `yaml:",inline"`
+// Config is a gopass config handler.
+type Config struct {
+	root *gitconfig.Configs
+	cfgs map[string]*gitconfig.Configs
 }
 
-// New creates a new config with sane default values.
+// New initializes a new gopass config. It will handle legacy configs as well.
 func New() *Config {
-	return &Config{
-		AutoImport:    false,
-		ClipTimeout:   45,
-		ExportKeys:    true,
-		Mounts:        make(map[string]string),
-		Notifications: true,
-		Parsing:       true,
-		Path:          PwStoreDir(""),
-		ConfigPath:    configLocation(),
-	}
+	return newWithOptions(false)
 }
 
-// CheckOverflow implements configer. It will check for any extra config values not.
-// handled by the current struct.
-func (c *Config) CheckOverflow() error {
-	return checkOverflow(c.XXX)
+// NewNoWrites initializes a new config that does not allow writes. For use in tests.
+func NewNoWrites() *Config {
+	return newWithOptions(true)
 }
 
-// Config will return a current config.
-func (c *Config) Config() *Config {
+func newWithOptions(noWrites bool) *Config {
+	c := &Config{
+		cfgs: make(map[string]*gitconfig.Configs, 10),
```

Comment:
Yes, but most people won't have all of them set.

But it doesn't really matter. I guess we shouldn't use a magic constant here at all.
This will likely never play a performance relevant role - unless someone is creating tons of Config structs.